src/objects/DataImportBatch__c.object

```diff
@@ -110,6 +110,19 @@
         <type>LongTextArea</type>
         <visibleLines>3</visibleLines>
     </fields>
+    <fields>
+        <fullName>Batch_Number__c</fullName>
+        <description>The unique, auto-numbered identifier for this batch.</description>
+        <externalId>true</externalId>
+        <inlineHelpText>The unique, auto-numbered identifier for this batch.</inlineHelpText>
+        <label>Batch Number</label>
+        <length>30</length>
```

Comment:
I limited it to 30 characters because that is the length of custom fields of data type "Auto Number" on the platform.  There is no specific requirement for the length on the WI.  After I push the next round of changes, the display format text will be limited to 23 characters (including the mandatory "{0}") and the starting number will be limited to 9 characters, to allow for incrementing and to fit inside the 30 char limit.  This would allow up to 20 char for the prefix value and up to 999,999,999 for the highest starting number.

Reply:
Just to follow up, we do have a requirement on the WI that the prefix+number should not exceed 30 char.

Reply:
I think the only part of this I'm struggling with is how 23 + 9 = 30 ... I will re-read the commit message you link Ken to see if I can figure it out!

Reply:
Ohhhhh wait because the curly brackets (and at least one of the zeroes) don't render in the final autonumber screen!